OCPBUGS#44579: prevent unintended unsupported sys config (#87660)
openshift/openshift-docs: modules/microshift-embed-ostree-enable-eus-repos.adoc

// Module included in the following assemblies:
//
// * microshift_install_rpm_ostree/microshift-embed-into-rpm-ostree.adoc

:_mod-docs-content-type: PROCEDURE
[id="microshift-enable-eus-repos_{context}"]
= Enabling extended support repositories for image building

If you have an extended support (EUS) release of {microshift-short}, you must enable the {op-system-base-full} EUS repositories for image builder to use. If you do not have an EUS version, you can skip these steps.

.Prerequisites

> * You have an EUS version of {microshift-short} or are updating to one.

Reviewer: > or are updating to one

Do we need this in the intro paragraph, too?  So, if folks who are updating to EUS are scanning the docs, they will see it.

> * You have cluster-administrator access.

Reviewer 2: ```suggestion
* You have root-user access to your build host.
```

.Procedure

. Update the `baseos` source by modifying the `/etc/osbuild-composer/repositories/rhel-9.4.json` file with the following values:
+
[source,terminal]
----
# ...
"baseurl": "https://cdn.redhat.com/content/eus/rhel<9>/<9.4>//baseos/os", # <1>
# ...
----

> <1> Replace _<9>_ with the major {op-system-base} version you are using, and replace _<9.4>_ with the _<major.minor>_ version. Be certain that the {op-system-base} version you choose is compatible with the {microshift-short} version you are using.

Reviewer: ??
```suggestion
<1> Replace _<9>_ with the major {op-system-base} version you are using, and replace _<9.4>_ with the _<major.minor>_ version. Be certain that the {op-system-base} version you choose is compatible with the {microshift-short} version you are using. For more information on version compatability, see.....
```

Reviewer: I think this is the more common wording, unless you think that "be certain" is stronger.
```suggestion
<1> Replace _<9>_ with the major {op-system-base} version you are using, and replace _<9.4>_ with the _<major.minor>_ version. Ensure that the {op-system-base} version you choose is compatible with the {microshift-short} version you are using.
```

> . Optional. Apply the `baseos` update by running the following command:

Reviewer: Per the [doc guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#callouts-in-code-blocks).
```suggestion
. Optional: Apply the `baseos` update by running the following command:
```

Author: Gah, always forget this one.

+
[source,terminal]
----
$ sudo sed -i "s,dist/rhel<9>/<9.4>/$(uname -m)/baseos/,eus/rhel<9>/<9.4>/$(uname -m)/baseos/,g" \
/etc/osbuild-composer/repositories/rhel-<9.4>.json # <1>
----

> <1> Replace _<9>_ with the major {op-system-base} version you are using, and replace _<9.4>_ with the _<major.minor>_ version. Be certain that the {op-system-base} version you choose is compatible with the {microshift-short} version you are using.

Reviewer: I think this is the more common wording, unless you think that "be certain" is stronger.
```suggestion
<1> Replace _<9>_ with the major {op-system-base} version you are using, and replace _<9.4>_ with the _<major.minor>_ version. Ensure that the {op-system-base} version you choose is compatible with the {microshift-short} version you are using.
```

> . Update the `appstream` source by modifying the `/etc/osbuild-composer/repositories/rhel-<major.minor>.json` with the following values:

Reviewer: ```suggestion
. Update the `appstream` source by modifying the `/etc/osbuild-composer/repositories/rhel-<major.minor>.json` file with the following values:
```

+
[source,terminal]
----
# ...
"baseurl": "https://cdn.redhat.com/content/eus/rhel<9>/<9.4>//appstream/os", # <1>
# ...
----

> <1> Replace _<9>_ with the major {op-system-base} version you are using, and replace _<9.4>_ with the _<major.minor>_ version. Be certain that the {op-system-base} version you choose is compatible with the {microshift-short} version you are using.

Reviewer: ```suggestion
<1> Replace _<9>_ with the major {op-system-base} version you are using, and replace _<9.4>_ with the _<major.minor>_ version. Ensure that the {op-system-base} version you choose is compatible with the {microshift-short} version you are using.
```

> . Optional. Apply the `apptream` update by running the following command:

Reviewer 2: ```suggestion
. Optional. Apply the `appstream` update by running the following command:
```

+
[source,terminal]
----
$ sudo sed -i "s,dist/rhel<9>/<9.4>/$(uname -m)/appstream/,eus/rhel<9>/<9.4>/$(uname -m)/appstream/,g" \
/etc/osbuild-composer/repositories/rhel-<9.4>.json # <1>
----

> <1> Replace _<9>_ with the major {op-system-base} version you are using, and replace _<9.4>_ with the _<major.minor>_ version. Be certain that the {op-system-base} version you choose is compatible with the {microshift-short} version you are using.

Reviewer: ```suggestion
<1> Replace _<9>_ with the major {op-system-base} version you are using, and replace _<9.4>_ with the _<major.minor>_ version. Ensure that the {op-system-base} version you choose is compatible with the {microshift-short} version you are using.
```

.Verification

> You can verify the the repositories by using `composer-cli` to display information about the source.

Reviewer 2: ```suggestion
You can verify the repositories by using `composer-cli` to display information about the source.
```

> You can verify the repositories by using `composer-cli` to display information about the source.

Reviewer: Nitpick, to ensure that folks will know what composer-cli is.
```suggestion
You can verify the repositories by using the `composer-cli` command to display information about the source.
```

. Verify the `baseos` source by running the following command:
+
[source,terminal]
----

> $ sudo composer-cli sources info appstream | grep 'url ='

Reviewer 2: ```suggestion
$ sudo composer-cli sources info baseos | grep 'url ='
```

----
.Example output
+
[source,text]
----

> url = "https://cdn.redhat.com/content/eus/rhel9/9.4/x86_64/appstream/os"

Reviewer 2: ```suggestion
url = "https://cdn.redhat.com/content/eus/rhel9/9.4/x86_64/baseos/os"
```

----

. Verify the `appstream` source by running the following command:
+
[source,terminal]
----

> $ sudo composer-cli sources info baseos | grep 'url ='

Reviewer 2: ```suggestion
$ sudo composer-cli sources info appstream | grep 'url ='
```

----
.Example output
+
[source,text]
----

> url = "https://cdn.redhat.com/content/eus/rhel9/9.4/x86_64/baseos/os"

Reviewer 2: ```suggestion
url = "https://cdn.redhat.com/content/eus/rhel9/9.4/x86_64/appstream/os"
```
